### ISSUE TITLE
Test walletconnect

### ISSUE THREAD

**Issue by mihaisc** [oceanprotocol/market#1148]
Test if portis/walletconnect(+ ledger) are working in v4 for all the basic flows: publish (all 3 pricing schemas) , order, trade,  add/remove liquidity 

Previous issues with walletconnect : https://github.com/oceanprotocol/market/issues/592

**Comment by LucaMilanese90** [comment]
I can't seem to be able to connect to the market with Portis, is it because https://v4.market.oceanprotocol.com/ is a preview link? 
![Screenshot 2022-03-17 at 16 38 09](https://user-images.githubusercontent.com/20192135/158838570-e1fbca8b-cd2b-419f-b1b8-2d41ea4f1bed.png)

**Comment by AnaLoznianu** [comment]
@kremalicious Can we add the Portis key?

**Comment by kremalicious** [comment]
Fun fact, 229 accounts seem to have used Portis in Ocean Market so far:

<img width="1043" alt="Screen Shot 2022-03-29 at 11 42 40" src="https://user-images.githubusercontent.com/90316/160594342-f7febe4f-a604-4a65-8d58-f86d2d4aef32.png">

**Comment by kremalicious** [comment]
Alright, set the allow list domains to:

```
https://market.oceanprotocol.com,https://*.oceanprotocol.vercel.app,https://*.market.oceanprotocol.com
```

**Comment by kremalicious** [comment]
Regression of combination walletconnect + Ledger Live in v4 preview:

On iOS and iPadOS getting only the QR code after selecting walletconnect when it should bring up the mobile app screen to use it on same device, like on balancer:

![image](https://user-images.githubusercontent.com/90316/160603383-ba178c67-8e5f-4267-902e-939506a3ae74.png)

We want this:
![image](https://user-images.githubusercontent.com/90316/160603418-e4e58e0c-3bf2-49f1-8ce3-12949fb625f9.png)

**Comment by mihaisc** [comment]
Interesting last time i tested i explicitly remember that cool walletconnect select popup, i was kinda impressed. Not sure what happened , i don't think we touched the integration since then

**Comment by LucaMilanese90** [comment]
I have been able to successfully publish an asset from a wallet connected through WalletConnect scanning the QR through Metamask Mobile. Correctly received the transaction notifications and sent the confirmation from my smartphone (iOS).
https://v4.market.oceanprotocol.com/asset/did:op:6137dd7a9d068708df3b9527ebb3a82420c21c49eed36e28ffb067053839fa54

The connection works also with Rainbow mobile (through WalletConnect) but I was not able to select a testnet so I couldn't try if the publish flow works. 
I noticed though that the nft image and infos are no longer displayed correctly on OpenSea.

**Comment by kremalicious** [comment]
great, thanks for testing!

> I was not able to select a testnet 

It's a bit hidden and weird interaction, but in Rainbow mobile in settings at very bottom is _Developer Settings_, there you can activate and switch to testnets.

<img src="https://user-images.githubusercontent.com/90316/161993752-5b7df824-3fc3-4fc4-b720-ca17367d4c01.PNG" width="300"/>

Last time I used Portis there was also some setting somewhere to show testnets in the list.

> I noticed though that the nft image and infos are no longer displayed correctly on OpenSea.

yes, was hoping this just goes away but seems not, started tracking this in #1330

**Comment by kremalicious** [comment]
Needs another test round as we updated the way we do signatures for download, and WalletConnect has more priority than Portis.

**Comment by mihaisc** [comment]
I say we just drop portis support for now. We clearly don't have time to implement various wallet sdks.

**Comment by kremalicious** [comment]
I was about to kick out Portis in #1355, but then found a bunch of custom components commented out and just enabled them and this network switcher actually works:

<img width="359" alt="Screen Shot 2022-04-25 at 12 53 52" src="https://user-images.githubusercontent.com/90316/165075940-d4241e5e-4f07-468b-b467-16341b6fd00e.png">

Working meaning I can select Rinkeby, then click _Show Portis_ and I see that my network changed on Portis. But app does not detect it although we have some `handlePortisNetworkChange()` in there which most likely just have to be moved to our Web3 context provider. So if it's just this simple change to make it work we should run with it, otherwise we should kick it out as it would be only half-working

**Comment by LucaMilanese90** [comment]
I'm having troubles to download datasets after completing the purchase, at the signature time I always get this message (connected through walletconnect and metamask mobile
![Screenshot 2022-04-26 at 10 04 27](https://user-images.githubusercontent.com/20192135/165253603-1bcf6c48-7240-45ce-88fa-457f4596c0b2.png)
)

**Comment by LucaMilanese90** [comment]
Also buying assets with pool prices is not working correctly: 
- the loading message no longer appears while I'm approving transactions on the phone (I'm able to click multiple time on buy while the transactions are going on)
- After confirming all the transactions I see a price increase
- for a few seconds I saw the download button but then it went away before I could click on it
- the signature at the end fails

I have no way of recording what's happening on the phone, but I'm basically confirming a bunch of transactions, on the video though you can see at 00:49 and 1:59 the price jumps and the download button disappearing (I'm hovering on top with the cursor but I haven't click on it).

https://user-images.githubusercontent.com/20192135/165255954-a1276264-c441-496c-9c40-047ef90b3e70.mov

**Comment by LucaMilanese90** [comment]
> I'm having troubles to download datasets after completing the purchase, at the signature time I always get this message (connected through walletconnect and metamask mobile). This happens both for pool and free assets, I couldn't test on fixed priced assets because I'm having this #1356 problem on basically all the available datasets ![Screenshot 2022-04-26 at 10 04 27](https://user-images.githubusercontent.com/20192135/165253603-1bcf6c48-7240-45ce-88fa-457f4596c0b2.png)

On MM mobile connected through walletconnect I'm still having the same issue as of today: I can buy an asset but when it's time to download the operation fails with the same error.
The same happens for a compute job, it fails at the moment of signing but this time the errors are only in the console
![Screenshot 2022-06-28 at 09 47 28](https://user-images.githubusercontent.com/20192135/176123685-b8ef7487-aace-4cb9-b8f9-c986d84fd22f.png)

**Comment by mihaisc** [comment]
i think we can close this
